### ISSUE TITLE
EML exportation very slow

### ISSUE THREAD

**Issue by pirli54** [thunderbird/import-export-tools-ng#606]
Hi,

since the 128v, it seems that the EML exportation is very slow for big folders (for example folder with 1.100 items).  The .mbox function also seems to take a unsual long time when I compare to the 115v.

**Comment by pirli54** [comment]
It seems that the progamm is duplicating in such way the size/value of the items. ImportExport has found 
![Screenshot from 2024-07-19 13-59-03](https://github.com/user-attachments/assets/e8f7f38b-1a6f-4b01-bf4b-3ea2f025393b) more gigabits with a few emails than the folder is actually containing

**Comment by cleidigh** [comment]
@pirli54 
 Thanks for the post. So I am beginning to think this may be a profile /msg database issue. I have some comparisons on an imap folder with 1552 messages, more importantly it's 50MB total data size mbox. 

Exporting to mbox - 50MB/1552 messages 
v115 :  1.26s
v128 :  1.32s

Export same folder to eml messages 
v115 :  ~2s ( don't have timer)
v128 :  ~2+ s

Windows 10, 5 year old i7, Samsung 800 ssd

Now you would not be posting if you were getting numbers like these. Do you have comparison numbers. Get total folder size since message count is not accurate 

@cleidigh

**Comment by pirli54** [comment]
Hi, here are the results from my test:

Machine: Ubuntu 24.04, i7 on HDD, 8GB memory

Database: 168 e-mails (in one folder) - 27.2MB input

v115 (DEB): Exporting time to EML = 2sec with the same size in the output folder

v128 (SNAP): Exporting time to EML =30sec with a 2.1GB size (!) in the output folder

**Comment by cleidigh** [comment]
@pirli54 
Yes it's got to be a Thunderbird Linux issue where the read of the raw message is returning message sizes completely different and larger than reality. I will report up the chain, nothing I can do in IETNG unfortunately.
@cleidigh

**Comment by cleidigh** [comment]
@pirli54 
If you are interested, I can add debug on message sizes that might give the development team more information to solve this.
@cleidigh

**Comment by pirli54** [comment]
@cleidigh, yes thx it would be appreciated :)

in the meantime I will use TB115 or Windows as workaround

**Comment by pirli54** [comment]
Hi @cleidigh , I just wanted to inform you that I also opened a case on bugzilla. Unfortunetly, the situation remains unchanged until now.

https://bugzilla.mozilla.org/show_bug.cgi?id=1909102

**Comment by cleidigh** [comment]
@pirli54 
Have you tried my latest beta? It has some changes that I wonder might affect this.
https://github.com/thunderbird/import-export-tools-ng/blob/v14.1.3/xpi/beta/import-export-tools-ng-14.1.3-b3-tb.xpi

@cleidigh

**Comment by pirli54** [comment]
Hello @cleidigh 

please apologize for the delay.

I just tried the latest beta and I did not encounter any issue exporting the EML! The size is equivalent to the import files and the export time was as fast as I used to know.

Thx for you work.

What should I do concerning the bugzilla case? Report it as fixed?
